Add Testing section to the PR template (#13166)
bevyengine/bevy: .github/pull_request_template.md

- Describe the solution used to achieve the objective above.

## Testing

> - Did you test these changes? If so, how have you tested them? Are there any parts that need more testing?

Reviewer: ```suggestion
- Did you perform any manual tests for these changes? If so, how?
- Are there any parts that need more manual testing?
```

> - Is there anything people need to know when testing these changes?

Reviewer: ```suggestion
- Is there anything people need to know when manually testing these changes?
```

> - Is there anything people need to know when testing these changes?

Reviewer 2: ```suggestion
- How can other people (reviewers) test your changes? Is there anything specific they need to know?
```
